docs/support/tutorials/nextflow-puhti.md

```diff
@@ -1,191 +1,302 @@
-nextflow run /scratch/project_xxx/What_the_Phage/phage.nf --fasta /scratch/project_xxx/What_the_Phage/test-data/OX2_draft.fa --cores 4 --output results -profile local,singularity --cachedir /scratch/project_xxx/What_the_Phage/singularity --databases /scratch/project_xxx/What_the_Phage/databases/WtP_databases --workdir /scratch/project_xxx/What_the_Phage/workflow-phages-username 
+nextflow run ${SCRATCH}/phage.nf \
+         -profile local,singularity \
+         --fasta ${SCRATCH}/test-data/OX2_draft.fa \
+         --cores 4 \
+         --output results \
+         --cachedir ${SCRATCH}/singularity \
+         --databases ${SCRATCH}/databases/WtP_databases \
+         --workdir ${SCRATCH}/workflow-phages-username
 ```
 
-Please note that the Singularity images in this example should be in the folder:
-`"/scratch/project_xxx/What_the_Phage/singularity"`. Otherwise, Nextflow tries to download 
-the images again which increases the likelihood of failures. Remember to edit `project_xxx`
-to match your computing project.
+Please note that the container images in this example should be in the folder
+`/scratch/<project>/What_the_Phage/singularity`. Otherwise, Nextflow tries to
+download the images again which increases the likelihood of failures. Remember
+to edit `<project>` to match your actual CSC computing project name.
+
+## 7. Demonstration of nf-core Nextflow pipeline using HyperQueue executor (optional)
+
+In this example, let's use the
+[HyperQueue meta-scheduler](../../apps/hyperqueue.md) for executing a Nextflow
+pipeline. This executor can be used to scale up analysis across multiple nodes
+when needed.
+
+Here is a batch script for running a [nf-core pipeline](https://nf-co.re/pipelines) on Puhti:
+
+```bash
+#!/bin/bash
+#SBATCH --job-name=nextflowjob
+#SBATCH --account=<project>
+#SBATCH --partition=small
+#SBATCH --time=00:10:00
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=40
+#SBATCH --mem-per-cpu=2G
+  
+module load hyperqueue/0.16.0
+module load nextflow/22.10.1
+
+# create a directory on scratch folder for running nextflow pipeline
+export SCRATCH=/scratch/<project>/$USER/nextflow
+mkdir -p ${SCRATCH}
+cd ${SCRATCH}
+
+export APPTAINER_TMPDIR="${SCRATCH}"
+export APPTAINER_CACHEDIR="${SCRATCH}"
+unset XDG_RUNTIME_DIR
+
+# Specify a location for the HyperQueue server
+export HQ_SERVER_DIR=${PWD}/hq-server-${SLURM_JOB_ID}
+mkdir -p "${HQ_SERVER_DIR}"
+
+# Start the server in the background (&) and wait until it has started
+hq server start &
+until hq job list &>/dev/null ; do sleep 1 ; done
+
+# Start the workers in the background and wait for them to start
+srun --overlap --cpu-bind=none --mpi=none hq worker start --cpus=${SLURM_CPUS_PER_TASK} &
+hq worker wait "${SLURM_NTASKS}"
+
+# As an example, let's clone a nf-core pipeline and run a test sample
+git clone https://github.com/nf-core/rnaseq.git -b 3.10
+cd rnaseq
+nextflow run main.nf -profile test,singularity --outdir . -resume
```

Comment:
As a suggestion, would this make sense to include to make the example batch script work more out-of-the-box?
```suggestion
cd rnaseq

# Ensure Nextflow uses the right executor and knows how much it can submit
echo "executor {
  queueSize = $(( 40*SLURM_NNODES ))
  name = 'hq'
  cpus = $(( 40*SLURM_NNODES ))
}" >> nextflow.config

nextflow run main.nf -profile test,singularity --outdir . -resume
```

Reply:
The suggested change would work here. Please add git module and increase runtime to be half an hour or even more (sometimes, pulling images would take more time than usual)